### PR TITLE
Create Searching-algorithm.md

### PR DESCRIPTION
**Pull request by Bhumika-00**
Added searching algo

### Description
Searching Algorithms are methods used to find a specific value or condition within a collection of data. They are fundamental in computer science and software development, enabling efficient data retrieval from various data structures. 

Fixes # (issue number)

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/basic-dsa/array/Searching-algorithm.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        mid = (left + right) / 2
+        if A[mid] == target then
+            return mid
+        else if A[mid] < target then
+            left = mid + 1
+        else
+            right = mid - 1
+    end while
+    return -1
+end procedure
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/Searching-algorithm.md:38]: replace your code 

```
procedure binarySearch(A : sorted list of items, target)
    left = 0
    right = length(A) - 1
    while left <= right do
        mid = (left + right) / 2
        if A[mid] == target then
            return mid
        else if A[mid] < target then
            left = mid + 1
        else
            right = mid - 1
    end while
    return -1
end procedure
```

with

````md
```
procedure binarySearch(A : sorted list of items, target)
    left = 0
    right = length(A) - 1
    while left <= right do
        mid = (left + right) / 2
        if A[mid] == target then
            return mid
        else if A[mid] < target then
            left = mid + 1
        else
            right = mid - 1
    end while
    return -1
end procedure
```
````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
Please read our Contribution Guidelines and our previous document structure. If you have any questions, please don't hesitate to ask.

**Comment by Bhumika-00** [comment]
Okay

**Comment by Bhumika-00** [comment]
Please review #581 since it was accepted and contains same data with
replacement that you are asking for

**Pull request closed without merging**